### PR TITLE
5 user roles

### PR DESCRIPTION
**Pull request by brockcp**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
-  # :confirmable, :lockable, :timeoutable and :omniauthable
+
+  enum role: [:standard, :premium, :admin]
+  after_initialize :set_default_role, :if => :new_record?
+
+  def set_default_role
+    self.role ||= :standard
+  end
+
+  has_many :wikis #, dependent: :destroy
```
Comment (durhamka) [inline app/models/user.rb:10]: Curious, why you commented the dependent: destroy out?

File: app/policies/application_policy.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    #false #false is default
+    user.present?
+  end
+
+  def new?
+    create?
+  end
+
+  def update?
+    #false #false is default
```
Comment (durhamka) [inline app/policies/application_policy.rb:27]: You would want to remove these comments if committing in real life.

File: app/policies/wiki_policy.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  def admin_list?
+    user.admin?
+  end
+
+   def create?
+     user.present?
+   end
+
+   def update?
+     user.present? #CURRENT
```
Comment (durhamka) [inline app/policies/wiki_policy.rb:23]: Wondering what you were doing here?

File: app/views/wikis/show.html.erb [hunk 4]
```diff
@@ -1,6 +1,26 @@
 <h1><%= @wiki.title %></h1>
-<h1><%= @wiki.body %></h1>
 
-<%= link_to "EDIT WIKI", edit_wiki_path(@wiki), class: 'btn btn-warning' %>
-<%= link_to "NEW WIKI", new_wiki_path(@wiki), class: 'btn btn-success' %>
-<%= link_to "DESTROY", @wiki, method: :delete, class: "btn btn-danger", data: { confirm: 'Are you sure?' } %>
+  <% if (@wiki.private && current_user && current_user!=@wiki.user) ||
```
Comment (durhamka) [inline app/views/wikis/show.html.erb:3]: Nice! You could extract this into a helper if you wanted to make this more readable.

**Review by durhamka — APPROVED** [review]
Great job, Brock! Left a couple comments for you. This is a long one!

File: app/models/user.rb (hunk 1 above)
Comment (brockcp) [reply app/models/user.rb:10]: I was testing a bunch of different configurations in order to see the results and thus better understand the subject matter, and simply didnt get back to uncomment this code.

File: app/policies/application_policy.rb (hunk 2 above)
Comment (brockcp) [reply app/policies/application_policy.rb:27]: I certainly would never leave this kind of comment in real development. this is just for my learning purposes.

File: app/policies/wiki_policy.rb (hunk 3 above)
Comment (brockcp) [reply app/policies/wiki_policy.rb:23]: Again, I was testing different configurations. The checkpoint called for establishing 3 separate roles and with a private feature for the wikis. I kind of jumped the gun and tried to figure out how to implement functionality for these, which will be called for in later checkpoints.

File: app/views/wikis/show.html.erb (hunk 4 above)
Comment (brockcp) [reply app/views/wikis/show.html.erb:3]: I think I need to better understand these core functions before moving on to wrap them in a helper.

**Pull request merged**